### ISSUE TITLE
index merge join hang when call `Close function

### ISSUE THREAD

**Issue by lzmhhh123** [pingcap/tidb#18068]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
executes twice
```sql
SELECT /*+ INL_JOIN(uc) */ 1 FROM user_class uc JOIN class c ON (uc.class_id = c.class_id AND c.state = '1') WHERE c.school_id = 120052 LIMIT 0, 1000;
```

### 2. What did you expect to see? (Required)
Both queries get the right results.

### 3. What did you see instead (Required)
The second query hangs.
![image](https://user-images.githubusercontent.com/26040259/84861095-e4eab900-b0a2-11ea-81eb-2cc741d91a9e.png)

### 4. Affected version (Required)
v4.0

### 5. Root Cause Analysis
Index merge join can't close the outer worker.

**Comment by XuHuaiyu** [comment]
reproduce step:
1. change the source code
``` diff
--- a/executor/index_lookup_merge_join.go
+++ b/executor/index_lookup_merge_join.go
@@ -217,7 +217,7 @@ func (e *IndexLookUpMergeJoin) newOuterWorker(resultCh, innerCh chan *lookUpMerg
                executor:              e.children[0],
                resultCh:              resultCh,
                innerCh:               innerCh,
-               batchSize:             32,
+               batchSize:             1,
                maxBatchSize:          e.ctx.GetSessionVars().IndexJoinBatchSize,
                parentMemTracker:      e.memTracker,
                nextColCompareFilters: e.lastColHelper,
diff --git a/sessionctx/variable/varsutil.go b/sessionctx/variable/varsutil.go
index 4e630910b..0573b2ab7 100644
--- a/sessionctx/variable/varsutil.go
+++ b/sessionctx/variable/varsutil.go
@@ -298,7 +298,7 @@ const (
        // initChunkSizeUpperBound indicates upper bound value of tidb_init_chunk_size.
        initChunkSizeUpperBound = 32
        // maxChunkSizeLowerBound indicates lower bound value of tidb_max_chunk_size.
-       maxChunkSizeLowerBound = 32
+       maxChunkSizeLowerBound = 1
 )

```

2. run the following sql

``` sql
create table t(a int, index(a));
create table s(a int, index(a));

insert into t values(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1);
insert into s values(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1),(1);

select  /*+ inl_merge_join(s)*/ 1 from t join s on t.a = s.a limit 1;
select  /*+ inl_merge_join(s)*/ 1 from t join s on t.a = s.a limit 1; -- This SQL will be stuck.
```

**Comment by zbingwen** [comment]
hi @lzmhhh123  
which version is this bug fixed? I encountered the same phenomenon in 5.7.25-TiDB-v4.0.6.
git hash is 51d365fc45fdfc039eb204a96268c5bd1c55075f.

**Comment by zbingwen** [comment]
![image](https://user-images.githubusercontent.com/6135275/102036724-8490c600-3dfe-11eb-9f19-cafa98a92986.png)
